pymc/step_methods/metropolis.py

```diff
@@ -257,14 +257,15 @@ def astep(self, q0: RaveledVars) -> Tuple[RaveledVars, StatsType]:
             q = floatX(q0d + delta)
 
         if self.elemwise_update:
+            q0d = q0d.copy()
             q_temp = q0d.copy()
             # Shuffle order of updates (probably we don't need to do this in every step)
             np.random.shuffle(self.enum_dims)
             for i in self.enum_dims:
                 q_temp[i] = q[i]
                 accept_rate_i = self.delta_logp(q_temp, q0d)
                 q_temp_, accepted_i = metrop_select(accept_rate_i, q_temp, q0d)
-                q_temp[i] = q_temp_[i]
+                q_temp[i] = q0d[i] = q_temp_[i]
```

Comment:
If I understand correctly, this basically means that you are performing Gibbs updates at each batch dimension iteratively. So if you have a vector variable $\vec{v}$ and a proposal vector $\vec{u}$, you will first see if you accept or reject the vector $(u_{0}, v_{1}, ..., v_{n})$ versus $\vec{v}$. If you accept this proposal, the next proposal will be $(u_{0}, u_{1}, v_{2}, ..., v_{n})$ versus $(u_{0}, v_{1}, ..., v_{n})$, and so on. If I understand correctly, the past behavior was doing the iterative element wise updates but always compared against $\vec{v}$.
If I my understanding is correct, then this looks ok to me